zerver/tests/test_webhooks_common.py

```diff
@@ -44,6 +50,32 @@ def test_webhook_http_header_header_does_not_exist(self) -> None:
         self.assertEqual(msg.sender.email, notification_bot.email)
         self.assertEqual(msg.content, expected_message)
 
+    def test_notify_bot_owner_on_invalid_json(self)-> None:
+        @api_key_only_webhook_view('ClientName')
+        def my_webhook_raises_exception(request: HttpRequest, user_profile: UserProfile) -> None:
+            raise InvalidJSONError("Malformed JSON")
+
+        @api_key_only_webhook_view('ClientName', notify_bot_owner_on_invalid_json=True)
+        def my_webhook(request: HttpRequest, user_profile: UserProfile) -> None:
+            raise InvalidJSONError("Malformed JSON")
+
+        webhook_bot_email = 'webhook-bot@zulip.com'
+        webhook_bot_realm = get_realm('zulip')
+        webhook_bot = get_user(webhook_bot_email, webhook_bot_realm)
+        webhook_bot_api_key = get_api_key(webhook_bot)
+        request = HostRequestMock()
+        request.POST['api_key'] = webhook_bot_api_key
+        request.host = "zulip.testserver"
+
+        with self.assertRaisesRegex(JsonableError, "Malformed JSON"):
+            my_webhook_raises_exception(request)  # type: ignore # mypy doesn't seem to apply the decorator
+
```

Comment:
This test doesn't verify that with the setting disabled it doesn't send a message, and thus doesn't verify one of ten most important aspects of the logic (that is the reason why to test both cases).  I fixed this before merging.